### PR TITLE
Add burn contract to tests

### PR DESCRIPTION
**Pull request by begmaroman**
# Description

Update tests according to EIP-1559 changes made before. New tests will be added in the next PR.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

File: state/executor.go [hunk 1]
```diff
@@ -574,7 +578,7 @@ func (t *Transition) apply(msg *types.Transaction) (*runtime.ExecutionResult, er
 	// We use EIP-1559 fields of the tx if the london hardfork is enabled.
 	// Effective tip be came to be either gas tip cap or (gas fee cap - current base fee)
 	effectiveTip := new(big.Int).Set(gasPrice)
-	if t.config.London && msg.Type != types.StateTx {
+	if t.config.London && msg.Type == types.DynamicFeeTx {
```
Comment (goran-ethernal) [inline state/executor.go:581]: There is a condition in this `apply` function:

```
// Burn some amount if the london hardfork is applied.
// Basically, burn amount is just transferred to the current burn contract.
if t.config.London && msg.Type != types.StateTx {
```

Should we change it to check if `msg.Type == types.DynamicFeeTx` like we did here? Maybe we can move it in this if block.

File: types/rlp_encoding_test.go [hunk 2]
```diff
@@ -281,53 +296,59 @@ func TestRLPMarshall_And_Unmarshall_TxType(t *testing.T) {
 func testRLPData(arena *fastrlp.Arena, ommitValues map[string]bool) []byte {
 	vv := arena.NewArray()
 
-	_, ommit := ommitValues["Nonce"]
-	if !ommit {
+	if ommit, _ := ommitValues["ChainID"]; !ommit {
```
Comment (vcastellm) [inline types/rlp_encoding_test.go]: I hate to be the nitpicker here but it's `omit` and `omitted` 😄

**Review by vcastellm — COMMENTED** [review]
Minor comment

File: types/rlp_encoding_test.go (hunk 2 above)
Comment (begmaroman) [reply types/rlp_encoding_test.go]: Fixed

File: state/executor.go (hunk 1 above)
Comment (begmaroman) [reply state/executor.go:581]: The idea here is that there are not gas fee cap and gas tip cap in non-dynamic-gas-fee tx so it jus does not make any sense to apply this check for legacy or state txs.

**Review by vcastellm — APPROVED** [review]

**Pull request merged**